doc/source/user-experience.rst

```diff
@@ -0,0 +1,282 @@
+   by editing the ``config.yaml`` file that we originally created to include
+   the jupyter hashes. Edit ``config.yaml`` by including these lines in it:
+
+    .. code-block:: bash
+
+          singleuser:
+            image:
+              name: gcr.io/<project-name>/<image-name>
+              tag: <tag>
+
+7. **Tell helm to update JupyterHub to use this configuration.** Using the
+   normal method to `apply the change <#applying-configuration-changes>`_ to
+   the config.
+8. **Restart your notebook if you are already logging in** If you already have
+   a running JupyterHub session, you’ll need to restart it (by stopping and
+   starting your session from the control panel in the top right). New users
+   won’t have to do this.
+9. **Enjoy your new computing environment!** You should now have a live
+   computing environment built off of the Docker image we’ve created.
+
+   .. note::
+      The contents of your GitHub repository might not show up if you have
+      enabled persistent storage. Disable persistent storage if you want them
+      to show up!
+
+Setting environment variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Another way to affect your user's environment is by setting
+`environment variables <https://en.wikipedia.org/wiki/Environment_variable>`_.
+You can set them up in your Docker image too, but it is sometimes
+easier to set them up in your helm chart!
+
+As usual, you would just edit your ``config.yaml`` file and re-apply!
+
+.. code-block:: yaml
+
+   singleuser:
+     extraEnv:
+       EDITOR: "vim"
+
+This will set the environment variable ``EDITOR`` to the value ``vim``. You
+can set any number of static environment variables here as you want.
+
+Users can read the environment variables in their code in various ways.
+
+In python,
+
+.. code-block:: python
+
+   import os
+   my_value = os.environ["MY_ENVIRONMENT_VARIABLE"]
+
+Other languages will have their own methods of accessing this.
+
+User resources
+--------------
+
+User resources are the CPU / RAM / Storage resources you provide your users.
+
+
+Setting memory and CPU guarantees / limits for your users
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each user on your JupyterHub gets a slice of memory and CPU to use. There are
+two ways to specify how much users get to use: resource *guarantees* and
+resource *limits*.
+
+A resource *guarantee* means that all users will have *at least* this resource
+available at all times, but they may be given more resources if they're
+available. For example, if users are *guaranteed* 1G of RAM, users can
+technically use more than 1G of RAM if these resources aren't being used by
+other users.
+
+A resource *limit* sets a hard limit on the resources available. In the example
+above, if there were a 1G memory limit, it would mean that users could use
+no more than 1G of RAM, no matter what other resources are being used on the
+machines.
```

Comment:
Maybe a quick note that "if users used more than 1GB of RAM, their kernel would be restarted automatically". Or some note to the effect of saying what the subjective experience would be from the user's perspective